etl/steps/data/garden/survey/2023-08-04/trust_surveys.py

```diff
@@ -0,0 +1,46 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+from typing import cast
+
+import owid.catalog.processing as pr
+from owid.catalog import Dataset
+
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow_latino = cast(Dataset, paths.load_dependency("latinobarometro_trust"))
+
+    # Read table from meadow dataset.
+    tb_latino = ds_meadow_latino["latinobarometro_trust"].reset_index()
+
+    # Load meadow dataset.
+    ds_meadow_ess = cast(Dataset, paths.load_dependency("ess_trust"))
+
+    # Read table from meadow dataset.
+    tb_ess = ds_meadow_ess["ess_trust"].reset_index()
+
+    #
+    # Process data.
+
+    # Concatenate the two tables.
+    tb = pr.concat([tb_latino, tb_ess[["country", "year", "trust"]]], ignore_index=True, short_name="trust_surveys")
```

Comment:
Quick question, @pabloarosado (not to review the entire PR): I am using this `pr.concat` function to keep metadata from both sources. Though it seems that it keeps part of it, when I run the steps the ETL doesn't find sources, so for now I'm adding them in `trust_surveys.meta.yml`, which is not ideal. Is there anything I should change?

Reply:
Currently, `pr.concat` should ensure all columns keep their original licenses and sources (or origins). If a column is shared by multiple tables concatenated, then that column should have multiple sources. If that is not happening, then there's a bug somewhere (please let me know).
Another topic is how to propagate additional metadata (e.g. `display` of the variables). Mojmir created [a PR](https://github.com/owid/etl/pull/1412) so that `pr.concat` automatically propagates all metadata of variables, but it has the issue that, if a column is shared by multiple tables, then only the metadata of the last table will prevail. In my opinion, we have to decide how to combine metadata in such cases. So, if this is your situation (you want to propagate all metadata fields, beyond licenses/sources/origins) then unfortunately you'll have to do it manually, for now.

Reply:
Thanks. When I only leave the dataset's title and description in the yaml file (commenting out the rest) the sources are not propagated:
<img width="1370" alt="image" src="https://github.com/owid/etl/assets/63430031/39973953-b19c-412a-8d9f-fb0e1294eab7">

And for the trust variable I have a small subset of the metadata, I lose the description and the display metadata but I keep the title and units (short and long)
<img width="390" alt="image" src="https://github.com/owid/etl/assets/63430031/a4e45ffe-6e4e-4418-8d85-33102792953f">

So for now I need to fill metadata again to get
<img width="1358" alt="image" src="https://github.com/owid/etl/assets/63430031/4df77945-0489-490f-adda-dd6cdae50d64">
<img width="1404" alt="image" src="https://github.com/owid/etl/assets/63430031/4d43e45e-34e3-489f-bf2d-cfc1754070a1">

Reply:
You are right that the dataset sources (the ones you see when doing `table.metadata.dataset`) are currently not properly propagated (I have a fix for that in an open PR), but, in the end, what matters the most is that each of the individual variables has the correct sources and licenses.
I'd need to look into the operations done to the trust variable, to see where the metadata is lost. But the `display` metadata is definitely not taken care of (yet), so yes, that's expected to be lost.
In a nutshell, yes, feel free to redefine the metadata manually. We'll need to look into these cases to see how we can improve metadata propagation in the future.